### ISSUE TITLE
Testing

### ISSUE THREAD

**Issue by paked** [paked/messenger#1]
I think it should be relatively easy to get some basic tests going, however it looks like it will be a pain to get any full blown integration tests happening.

**Comment by dgellow** [comment]
That would be great, yes! Do you already have a plan? I would be happy to contribute.

**Comment by paked** [comment]
Not really. I'll happily merge PRs with this stuff, but I don't have any plan worth mentioning.

**Comment by dgellow** [comment]
Ok, if I have some free time (and don't forget) I will make a few PRs to unit tests.
